mozregression/download_manager.py

```diff
@@ -137,6 +138,12 @@ def _update_progress(self, current, total):
                 self.__progress(self, current, total)
 
     def _download(self, url, dest, finished_callback, chunk_size, session):
+        # save the file under a temporary name
+        # this allow to not use a broken file in case things went really bad
+        # while downloading the file (ie the python interpreter is killed
+        # abruptly)
+        real_dest = dest
+        dest += '.tmp'
```

Comment:
I don't like redefining function arguments. I'd prefer to set a variable called "temp_dest" and assign dest + '.tmp'.